extensions/interactions/DragAndDropSortInput/directives/oppia-interactive-drag-and-drop-sort-input.directive.ts

```diff
@@ -93,9 +107,31 @@ angular.module('oppia').directive('oppiaInteractiveDragAndDropSortInput', [
               ctrl.dataMaxDepth = 1;
             }
 
-            // Make list of dicts from the list of choices.
-            for (var i = 0; i < ctrl.choices.length; i++) {
-              ctrl.list.push({title: ctrl.choices[i], items: []});
+            let savedSolution = (
+              $attrs.savedSolution !== undefined ?
+              JSON.parse($attrs.savedSolution) : []
+            );
+
+            if (savedSolution.length) {
+              // Pre populate with the saved solution, if present.
+              for (let contentIds of savedSolution) {
+                let item = {
+                  title: getHtmlOfContentId(contentIds[0]),
+                  items: []
+                };
+                for (let i = 1; i < contentIds.length; i++) {
+                  item.items.push({
+                    title: getHtmlOfContentId(contentIds[i]),
+                    items: []
+                  });
+                }
+                ctrl.list.push(item);
+              }
+            } else {
+              // Make list of dicts from the list of choices.
+              for (var i = 0; i < ctrl.choices.length; i++) {
```

Comment:
Why can't you use
```suggestion
              for (let choice of ctrl.choices) {
```
here?

Reply:
Done!